### PR TITLE
'listbirthdays' command

### PR DESCRIPTION
**Pull request by robo-mop**
First of all, let me just say I'm a huge fan of this bot, especially because I just can't remember my friend's birthdays no matter how hard I try XD

Me and my friends thought it'll be cool to have a feature where you could see the birthdays of everyone in the chat, so I went ahead and tried to add this command in the main.py file. Admittedly, I have limited experience with sql, but I doubt there are any problems with the code I added. However, I'd really appreciate if you could go over it once, and maybe point out the errors I've made if any, so I can learn from them.

Thank you so much!

### CODE REVIEW

**Comment by IronTony-Stark** [comment]
Hi, @robo-mop! 

Thank you for your kind words! You can’t imagine how happy I am to see that somebody not only liked the bot, but also contributed to this project 

I’m currently really short on time, so I may not be able to look at your feature in a week or two 😞 Sorry.. 😭 But once I got a free hour, I’ll check it out for sure! The idea is great 👍 

Thank you very much!

**Comment by FdelMazo** [comment]
Any chance you take a look at this @IronTony-Stark ? I would love this feature (and can help coding/reviewing)

File: main.py [hunk 1]
```diff
@@ -281,7 +281,58 @@ def answer(query_result):
...
+    cursor.execute(sql_query, message.chat.id)
+    results = cursor.fetchall()
+
+    for row in results:
+        username = get_username_or_first_name(message.chat.id, row[0])
+        month, day = row[1][:2], row[1][3:]
+        if month != current_month:
+            if current_month != -1:
+                list_of_birthdays += "\n\n"
+            month = current_month
```
Comment (IronTony-Stark) [inline main.py:321]: Must be the other way around: `current_month = month`

File: main.py [hunk 2]
```diff
@@ -281,7 +281,58 @@ def answer(query_result):
...
+        return
+
+    awake_mysql_db()
+
+    sql_query = "SELECT User_Id, DATE_FORMAT(Birthday, '%m.%d') FROM birthdays WHERE MONTH(Birthday) > -1 AND Chat_Id = %s ORDER BY MONTH(Birthday), Day(Birthday)"
+
+    list_of_birthdays = ""
+    current_month = -1
+    current_day = -1
+    cursor.execute(sql_query, message.chat.id)
```
Comment (IronTony-Stark) [inline main.py:312]: Won't work

`cursor.execute(sql_query, params)`
`params` is a **tuple**

So if you want to pass a single argument, you have to write `cursor.execute(sql_query, (param,))`

File: main.py [hunk 3]
```diff
@@ -281,7 +281,58 @@ def answer(query_result):
...
+    results = cursor.fetchall()
+
+    for row in results:
+        username = get_username_or_first_name(message.chat.id, row[0])
```
Comment (IronTony-Stark) [inline main.py:316]:
```
August
14 - @IronTony-Stark 
```

🔔 Everybody in the chat will receive notification when somebody calls this command

It's better to use user's first name

**Comment by IronTony-Stark** [comment]
Hello again, guys! @robo-mop @FdelMazo
Sorry it took me so long. Somehow I forgot about this merge request 😊 

First of all, great job! 🚀 
I love this feature and the code you've written is fine 
There are some important points I've mentioned above though. You can check [latest commit](https://github.com/IronTony-Stark/Telegram-Happy-Birthday-Bot/commit/c3568ca095a6a4be599579363d9ca48782883001) to see the fixes

Thanks again, I really appreciate your work 👍

**Comment by FdelMazo** [comment]
Thanks @IronTony-Stark  and @robo-mop !

**Pull request merged**